### PR TITLE
Implement static test runner

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/elastic/elastic-package/issues/226

This PR implements static test runner with its first responsibility - linting sample events.

### CODE REVIEW

**Comment by mtojek** [comment]
I know that the CI is failing, but the code should be reviewable.

**Comment by ycombinator** [comment]
> This PR implements static test runner with its first responsibility - linting sample events.

Just curious, do you have other static assets in mind for follow up PRs?

File: internal/fields/validate.go [hunk 1]
```diff
@@ -22,13 +22,23 @@ import (
 
 // Validator is responsible for fields validation.
 type Validator struct {
-	schema               []FieldDefinition
-	numericKeywordFields map[string]struct{}
+	schema []FieldDefinition
+
+	defaultNumericConvertion bool
```
Comment (ycombinator) [inline internal/fields/validate.go:27]: Spelling.

```suggestion
	defaultNumericConversion bool
```

File: internal/testrunner/runners/static/test_config.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	"github.com/elastic/elastic-package/internal/testrunner"
+)
+
+type testConfig struct {
+	testrunner.SkippableConfig `config:",inline"`
+}
+
+func newConfig(staticTestFolderPath string) (*testConfig, error) {
+	configFilePath := filepath.Join(staticTestFolderPath, "config.yml")
```
Comment (ycombinator) [inline internal/testrunner/runners/static/test_config.go:24]: Should we define this as part of the spec, as an optional file?

**Review by ycombinator — COMMENTED** [review]
Left a question and a minor typo fix suggestion.

File: internal/testrunner/runners/static/test_config.go (hunk 2 above)
Comment (mtojek) [reply internal/testrunner/runners/static/test_config.go:24]: You're right, I totally forgot about the package-spec.

BTW Should I wait with this PR until you'll merge https://github.com/elastic/elastic-package/pull/260 or do you prefer the other way round?

File: internal/fields/validate.go (hunk 1 above)
Comment (mtojek) [reply internal/fields/validate.go]: Thanks, fixed.

**Comment by mtojek** [comment]
> Just curious, do you have other static assets in mind for follow up PRs?

Maybe not in follow ups, but we don't have any mechanism to verify/control what's inside screenshots/icons, e.g. size, resources.

File: internal/testrunner/runners/static/test_config.go (hunk 2 above)
Comment (mtojek) [reply internal/testrunner/runners/static/test_config.go:24]: https://github.com/elastic/package-spec/pull/141

Comment (ycombinator) [reply internal/testrunner/runners/static/test_config.go:24]: I just merged #260 so you'll need to rebase this PR on top of `master`.

File: README.md [hunk 3]
```diff
@@ -150,23 +150,29 @@ _Context: package_
 
 Use this command to run tests on a package. Currently, there are two types of tests available.
 
+#### Asset Loading Tests
```
Comment (ycombinator) [inline README.md:153]: Thank you for adding this section.

File: README.md [hunk 4]
```diff
@@ -150,23 +150,29 @@ _Context: package_
...
 
 These tests allow you to exercise any Ingest Node Pipelines defined by your packages.
 
 For details on how to configure and run pipeline tests for a package, see the [HOWTO guide](docs/howto/pipeline_testing.md).
 
-#### System Tests
+#### Static Tests
 
-These tests allow you to test a package's ability to ingest data end-to-end. 
+These tests allow you to verify if all static resources of the package are valid, e.g. are all fields of the `sample_event.json` documented.
```
Comment (ycombinator) [inline README.md:167]:
```suggestion
These tests allow you to verify if all static resources of the package are valid, e.g. if all fields of the `sample_event.json` are documented.
```

File: README.md (hunk 3 above)
Comment (ycombinator) [reply README.md:153]: Never mind, I see this section was already there but is just re-sorted to the top. 😄

File: cmd/testrunner.go [hunk 5]
```diff
@@ -24,10 +24,16 @@ import (
...
 
+Asset Loading Tests
+These tests allow you to exercise installing a package to ensure that its assets are loaded into Elasticsearch and Kibana as expected.
+
 Pipeline Tests
 These tests allow you to exercise any Ingest Node Pipelines defined by your packages.
 For details on how to configure pipeline test for a package, review the HOWTO guide (see: https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md).
 
+Static Tests
+These tests allow you to verify if all static resources of the package are valid, e.g. are all fields of the sample_event.json documented.
```
Comment (ycombinator) [inline cmd/testrunner.go:35]:
```suggestion
These tests allow you to verify if all static resources of the package are valid, e.g. if all fields of the sample_event.json are documented.
```

File: docs/howto/asset_testing.md [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# HOWTO: Writing asset loading tests for a package
+# HOWTO: Running asset loading tests for a package
```
Comment (ycombinator) [inline docs/howto/asset_testing.md:1]: 👍

**Review by ycombinator — APPROVED** [review]
Left a couple minor doc language cleanup suggestions. After that, LGTM.

**Pull request merged**